### DISCUSSION TITLE
议视频语料标注应沿用原有.srt格式存储抑是转换为ELAN模式

### DISCUSSION THREAD

**Discussion by Guanchishan** [Nang-muoi-sing/cdo-corpus#34] [Ideas]
从[v0.3-bata.1](https://github.com/MindongLab/cdo-corpus/commit/402159455fc05973486e7f02480ef0496e7ec3c9)开始，我们[对视频语料采用`.srt`格式标记](https://web.archive.org/web/20200819195617/https://github.com/MindongLab/cdo-corpus/wiki/Subtitling-%E7%B7%A8%E5%AF%AB%E5%AD%97%E5%B9%95)，其中语料的metadata存储于对应`.srt`文件的前置Script Info中。

在2020年8月8日的榕典/输入法例会上，参与维护本库的人士商议依赖ELAN改写音频语料的标注。

然而已有的及未来的视频语料标注，应沿用原有.srt格式存储，抑是转换为ELAN模式，需要讨论其**必要性**与**实现难度**。

如果使用ELAN标注，需要起草新标注规范。

**Comment by ztl8702** [comment]
> 如果使用ELAN标注，需要起草新标注规范。

主要是规范一下，（如果很多人一起用ELAN的时候），大家使用ELAN给层起名的规范（有哪些layer，每个layer的含义）？

**Comment by Guanchishan** [comment]
已草規範:
https://github.com/MindongLab/cdo-corpus/wiki/Corpus-format-%E8%AA%9E%E6%96%99%E6%A0%BC%E5%BC%8F#%E5%88%86%E5%B1%A4%E6%A8%99%E8%A8%BB

Radium <notifications@github.com> 于2020年8月22日周六 下午9:56写道：

**Comment by Guanchishan** [comment]
> 如果很多人一起用ELAN的时候
btw目前音频都是单人说话。我还没想好多角色参与后怎么标记。

理想状况是能给分层加组，一组为一个角色。
但是目前不会加组，尝试给某层增加父层后，确认按钮就变灰了：
![image](https://user-images.githubusercontent.com/30859354/90961359-da9dce80-e4da-11ea-8199-2ae89180fa21.png)

我先再研究研究。

**Comment by Guanchishan** [comment]
> 但是目前不会加组，尝试给某层增加父层后，确认按钮就变灰了

在官方文档（[1](https://www.mpi.nl/corpus/html/elan/ch02.html)、[2](https://www.mpi.nl/corpus/html/elan/ch02s03.html#Sec_Adding_new_tier_types)）找到了解决方案，[升级了](https://github.com/MindongLab/cdo-corpus/wiki/Corpus-format-%E8%AA%9E%E6%96%99%E6%A0%BC%E5%BC%8F/9b34a4212a50e690f281c1b32b17d4984ee87415)分层标注规范。

**Comment by Guanchishan** [comment]
用srt的一個毛病就是，分詞標註、詞單位對譯等等基本無法實現，做不到如圖這麼精細地標註。![image](https://user-images.githubusercontent.com/30859354/91607593-b5abce80-e9a6-11ea-9764-7baf8989874a.jpeg)
srt根本上還是以句子爲單位的。

然而ELAN沒有什麼妥善的方案將語料metadata存於標註文件內。但是srt可以將metadata存於srt文件的前置字段內。
且，如果純音頻語料夠完成培養TTS這一目的的話，已有的srt就先不用費力去轉了。
但是換一個角度想，已有srt去轉換為elan的話，也許就能更早滿足培養TTS所需音頻總時長。

**Comment by ztl8702** [comment]
> 然而ELAN沒有什麼妥善的方案將語料metadata存於標註文件內

可以建一个同名的 .yaml文件

xxx.mp3
xxx.elan
xxx.yaml

这样

________________________________

**Comment by ztl8702** [comment]
现在先不要纠结格式。TTS训练数据有什么具体需求还没清晰。

________________________________

**Comment by Guanchishan** [comment]
暫時不需要解決此問題
